lighthouse-cli/bin.ts

```diff
@@ -219,21 +229,44 @@ function launchChromeAndRun(addresses: Array<string>,
   })
 }
 
-function lighthouseRun(addresses: Array<string>, config: Object, flags: Object) {
+function lighthouseRun(addresses: Array<string>,
+                       config: Object,
+                       flags: {
+                         excludeArtifacts: boolean,
+                         saveArtifacts: boolean,
+                         saveAssets: boolean,
+                       }) {
   // Process URLs once at a time
   const address = addresses.shift();
   if (!address) {
     return;
   }
 
   return lighthouse(address, flags, config)
+    .then((results: Results) => {
+      if (flags.saveArtifacts) {
+        return assetSaver.saveArtifacts(results.artifacts).then((_: Object) => results);
+      }
+      return results;
+    })
+    .then((results: Results) => {
+      if (flags.saveAssets) {
+        return assetSaver.saveAssets({url: results.url}, results.artifacts).then((_:Object) => results);
```

Comment:
`assetSaver.saveAssets` returns `Promise<undefined>`, so easier to do `.then(() => results)`